scripts/get_ec2_pricing.py

```diff
@@ -1,506 +1,56 @@
-    ec2_providers[Provider.EC2_US_WEST_OREGON] = ec2_providers['us-west-2']
-    ec2_providers[Provider.EC2_US_EAST] = ec2_providers['us-east-1']
-    ec2_providers[Provider.EC2_EU_CENTRAL] = ec2_providers['eu-central-1']
-
-    #formatting for easy copy/paste to mist.io/config.py
-    for provider in [Provider.EC2_EU_CENTRAL, Provider.EC2_EU_WEST, Provider.EC2_SA_EAST, Provider.EC2_AP_NORTHEAST1, Provider.EC2_AP_NORTHEAST2, Provider.EC2_AP_SOUTHEAST2, Provider.EC2_AP_SOUTHEAST, Provider.EC2_US_WEST, Provider.EC2_US_WEST_OREGON, Provider.EC2_US_EAST]:
-        print "        \"%s\": {" % provider
-        for key in ec2_providers[provider].keys()[:-1]:
-            print "            \"%s\": %s," % (key, json.dumps(ec2_providers[provider][key]))
-        key = ec2_providers[provider].keys()[-1]
-        print "            \"%s\": %s" % (key, json.dumps(ec2_providers[provider][key]))
-        #don't use a comma for the last key, for valid JSON
-        print '        },\n'
+class mydict(dict):
+    def __str__(self):
+        return json.dumps(self)
+
+csv_file = '/home/user/Desktop/unweb/awspricingfull/EXAMPLE_OUTPUT_AWS_FULL_RI_pricing.csv'
+
+regions = {}
+
+with open(csv_file, 'rb') as csvfile:
+    reader = csv.reader(csvfile, delimiter=',')
+    for row in reader:
+        instance_type = row[0]
+        if row[1] == 'ec2' and instance_type == 'ondemand':
+            region = row[2]
+            size = row[3]
+            os = row[6]
+            price = row[10]
+            if not regions.get(region):
+                regions[region] = {}
+            if not regions[region].get(size):
+                regions[region][size] = {}
+            regions[region][size][os] = price
+
+mist_regions = {}
+mist_regions[Provider.EC2_EU_WEST] = regions['eu-west-1']
+mist_regions[Provider.EC2_SA_EAST] = regions['sa-east-1']
+mist_regions[Provider.EC2_AP_NORTHEAST] = regions['ap-northeast-1']
+mist_regions[Provider.EC2_AP_SOUTHEAST2] = regions['ap-southeast-2']
+mist_regions[Provider.EC2_AP_SOUTHEAST] = regions['ap-southeast-1']
+mist_regions[Provider.EC2_US_WEST] = regions['us-west-1']
+mist_regions[Provider.EC2_US_WEST_OREGON] = regions['us-west-2']
+mist_regions[Provider.EC2_US_EAST] = regions['us-east-1']
+mist_regions[Provider.EC2_EU_CENTRAL] = regions['eu-central-1']
+mist_regions[Provider.EC2_AP_NORTHEAST2] = regions['ap-northeast-2']
+mist_regions[Provider.EC2_AP_NORTHEAST1] = regions['ap-northeast-1']
+
+# formatting for easy copy/paste to mist.io/config.py
+for provider in mist_regions:
+    print "        \"%s\": {" % provider
+    for key in mist_regions[provider].keys()[:-1]:
+        print "            \"%s\": %s," % (key, json.dumps(mist_regions[provider][key]))
+    key = mist_regions[provider].keys()[-1]
+    print "            \"%s\": %s" % (key, json.dumps(mist_regions[provider][key]))
+    # don't use a comma for the last key, for valid JSON
+    print '        },\n'
```

Comment:
Isn't there a nicer way to print these? Such as creating a single object and then dumping it to json?